pkg/asset/installconfig/gcp/validation_test.go

```diff
@@ -20,16 +20,20 @@ import (
 type editFunctions []func(ic *types.InstallConfig)
 
 var (
-	validNetworkName   = "valid-vpc"
-	validProjectName   = "valid-project"
-	invalidProjectName = "invalid-project"
-	validRegion        = "us-east1"
-	invalidRegion      = "us-east4"
-	validZone          = "us-east1-b"
-	validComputeSubnet = "valid-compute-subnet"
-	validCPSubnet      = "valid-controlplane-subnet"
-	validCIDR          = "10.0.0.0/16"
-	validClusterName   = "valid-cluster"
+	validNetworkName       = "valid-vpc"
+	validProjectName       = "valid-project"
+	invalidProjectName     = "invalid-project"
+	validRegion            = "us-east1"
+	invalidRegion          = "us-east4"
+	validZone              = "us-east1-b"
+	validComputeSubnet     = "valid-compute-subnet"
+	validCPSubnet          = "valid-controlplane-subnet"
+	validCIDR              = "10.0.0.0/16"
+	validClusterName       = "valid-cluster"
+	validShortPublicZone   = "valid-short-public-zone"
+	invalidShortPublicZone = "invalid-short-public-zone"
+	validLongPublicZone    = fmt.Sprintf("projects/%s/managedZones/%s", validProjectName, validShortPublicZone)
+	invalidLongPublicZone  = fmt.Sprintf("projects/%s/managedZones/%s", validProjectName, invalidShortPublicZone)
```

Comment:
I am gathering from this that we are allowing users to pass strings in two formats. I think this is another example of where it would be better solved by structuring the install config field as an object to accept a zone name and optional project id.